src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const groupedObj: GroupsMap<T> = {};
+
+  items.forEach((obj: T) => {
+    const property: string = `${obj[key]}`;
```

Comment:
Pls consider using a more descriptive variable name instead of 'obj'. For example, 'item' or 'element' would be more suitable.